packages/dd-trace/src/service-naming/extra-services.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const maxExtraServices = 64
+const extraServices = new Set()
+
+process.env.DD_EXTRA_SERVICES?.split(',')
```

Comment:
We are restricting the customer to inform these `extra-services` via environment variable, so they will cannot set these via `TracerOptions` when calling `tracer.init(tracerOptions)`. [doc](https://datadoghq.dev/dd-trace-js/interfaces/tracer.html#init)

I think this value should be managed via configuration, retrieved from environment variable or configuration object.

Reply:
(resolved the thread by accident sorry)
I agree, i don't know what is this habit of having env vars outside config.js nowadays but it shouldn't be done lightly, either we keep things in config.js or we discuss about a good reason not to.

Reply:
1. There already are a few entries in config.js that only accept env vars without programmatic options.
2. Same thing as above, there are a lot of config entries that are only used once. I'll also correct you here "this is the only place where the value is read **for now**"
3. I understand passing the config object everywhere is annoying, but either it's been architectured like that for a reason, or we need to change the architecture.

If you think the config system is well designed, then use it.
If you think the config system is not well designed, then you should propose to make it better, instead of ignoring it and straying away into your own path. We're all a team working on the same repo, if each of us decide to do their own architecture in their own corner, it will be a nightmare to maintain :/

Reply:
Actually you don't even have to pass the config object all the way here, you can just call registerExtraService in config.js, ezpz

Reply:
ah, for the record, I agree with you that it is generally better in `config.js`

Reply:
Alright well, ask what the rest of the guild think of it the, 3 possible solutions:
- read env var here
- pass the config all the way to this module
- call registerExtraServices() in config.js
I'll let you guys decide

Reply:
uops, I've just moved it to config.js and taken the opportunity to add the 'extraServices' property to [`TracerOptions`](https://github.com/DataDog/dd-trace-js/pull/3635/commits/39cbbe407b6fa49cc3a2b7556cb1fe619a07a7c2#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R276)